gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,233 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import static java.util.stream.Collectors.toCollection;
+
+import static uk.nhs.adaptors.pss.translator.util.DateFormatUtil.parseToInstantType;
+import static uk.nhs.adaptors.pss.translator.util.EncounterReferenceUtil.getEncounterReference;
+import static uk.nhs.adaptors.pss.translator.util.ResourceUtil.buildIdentifier;
+import static uk.nhs.adaptors.pss.translator.util.ResourceUtil.generateMeta;
+import static uk.nhs.adaptors.pss.translator.util.TextUtil.getLastLine;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.DiagnosticReport;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.InstantType;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.dstu3.model.Specimen;
+import org.hl7.v3.II;
+import org.hl7.v3.RCMRMT030101UK04Author;
+import org.hl7.v3.RCMRMT030101UK04Component02;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04CompoundStatement;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.TS;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.pss.translator.util.CompoundStatementUtil;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DiagnosticReportMapper {
+
+    private static final String EXTENSION_IDENTIFIER_ROOT = "2.16.840.1.113883.2.1.4.5.5";
+    private static final String META_PROFILE_URL_SUFFIX = "DiagnosticReport-1";
+    private static final String CLUSTER_CLASSCODE = "CLUSTER";
+    private static final String DR_SNOMED_CODE = "16488004";
+    private static final String SPECIMEN_CODE = "123038009";
+
+    private final CodeableConceptMapper codeableConceptMapper;
+    private final SpecimenMapper specimenMapper;
+    private final SpecimenCompoundsMapper specimenCompoundsMapper;
+
+    public List<DiagnosticReport> mapDiagnosticReports(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient,
+        List<Encounter> encounters, String practiceCode) {
+        var compositions = getCompositionsContainingClusterCompoundStatement(ehrExtract);
+        return compositions.stream()
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .flatMap(CompoundStatementUtil::extractAllCompoundStatements)
+            .filter(Objects::nonNull)
+            .filter(this::isDiagnosticReportCandidate)
```

Comment:
Could use the isDiagnosticReport filter in the ResourceFilterUtil? Might need to update it to how you are currently filtering